noir-projects/aztec-nr/aztec/src/macros/functions/utils.nr

```diff
@@ -376,6 +395,81 @@ pub(crate) comptime fn create_message_discovery_call() -> Quoted {
     }
 }
 
+/// Injects an authwit verification check of the form:
+/// ```
+///   if (!from.eq(context.msg_sender())) {
+///         assert_current_call_valid_authwit::<N>(&mut context, from);
+///     } else {
+///         assert(authwit_nonce, "Invalid auhtwit nonce. When 'from' and 'msg_sender' are the same, authwit_nonce must be zero");
+///     }
+/// ```
+/// where `from` and `authwit_nonce` are the names of the parameters that are expected to be present in the function definition.
+/// This check is injected by the `#[authorize_once("from_arg_name", "nonce_arg_name")]`, which allows the user to define
+/// which parameters to use.
+pub(crate) comptime fn create_authorize_once_check(f: FunctionDefinition) -> Quoted {
+    let name = f.name();
+    let maybe_authorize_once_args = AUTHORIZE_ONCE_REGISTRY.get(f);
+    if maybe_authorize_once_args.is_none() {
+        panic(
+            f"Functions marked with #[authorize_once] must have the #[private] or #[public] attribute placed last",
+        );
+    }
+    let mut from_arg_name = if maybe_authorize_once_args.unwrap().len() > 0 {
+        maybe_authorize_once_args.unwrap().pop_front().0
+    } else {
+        "from".as_ctstring()
+    };
```

Comment:
Jeeeez not sure I like this. Sure, in a token transfer it makes sense that `from` is the account to authorize, but this is way too implicit for my taste. If it was called `authorize_once_from` then _maybe_, but I'm really not a fan of this as it is. We can just remove the varargs and have this take the account that the authorization is for.

Reply:
What I implemented is essentially a "default" arg since they don't exist in noir, but I will change it.

Reply:
Reworked